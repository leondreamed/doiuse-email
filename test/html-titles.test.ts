import { describe, expect, test } from 'vitest';

import canIEmailJson from '../src/data/can-i-email.json';
import type { CanIEmailData } from '../src/types';
import {
  attributeTitles,
  elementAttributePairTitles,
  elementTitles
} from '../src/utils/titles/html';

const canIEmailData = canIEmailJson as unknown as CanIEmailData;
const allHTMLTitles = canIEmailData.data.filter((d) => d.category === 'html').map((d) => d.title);

const expectedElementTitles = [
  'address',
  '<audio> element',
  '<base>',
  '<bdi> element',
  '<blockquote> element',
  '<body> element',
  '<del> element',
  '<dialog> element',
  '<div> element',
  'HTML5 doctype',
  '<form> element',
  '<h1> to <h6> elements',
  'Image maps',
  '<link> element',
  '<ul>, <ol> and <dl>',
  '<marquee> element',
  '<meter> element',
  '<object> element',
  '<p> element',
  '<picture> element',
  '<progress> element',
  '<rp> element',
  '<rt> element',
  '<ruby> element',
  '<select> element',
  'HTML5 semantics',
  '<small> element',
  '<span> element',
  '<strike> element',
  '<strong> element',
  '<style> element',
  'Embedded <svg> image',
  '<table> element',
  '<textarea> element',
  '<video> element',
  '<wbr> element'
];

const expectedAttributeTitles = [
  'align attribute',
  'aria-describedby attribute',
  'aria-hidden attribute',
  'aria-label attribute',
  'aria-labelledby attribute',
  'aria-live attribute',
  'background attribute',
  'dir attribute',
  'height attribute',
  'lang attribute',
  'loading attribute',
  'required attribute',
  'role attribute',
  'srcset and sizes attributes',
  'target attribute',
  'valign attribute',
  'width attribute'
];

const expectedElementAttributePairTitles = [
  // <html amp4email>
  'AMP for Email',
  'Local anchors',
  '<button type="reset"> element',
  '<button type="submit"> element',
  '<input type="checkbox"> element',
  '<input type="hidden"> element',
  '<input type="radio"> element',
  '<input type="reset"> element',
  '<input type="submit"> element',
  '<input type="text"> element',
  'mailto: links'
];

function expectEqualTitles(titles: Record<string, unknown>, actualTitles: string[]) {
  expect(Object.keys(titles).sort()).toEqual(actualTitles.sort());
}

describe('css titles', () => {
  test('element titles', () => {
    expectEqualTitles(elementTitles, expectedElementTitles);
  });

  test('attribute titles', () => {
    expectEqualTitles(attributeTitles, expectedAttributeTitles);
  });

  test('element-attribute pair titles', () => {
    expectEqualTitles(elementAttributePairTitles, expectedElementAttributePairTitles);
  });

  test('all titles', () => {
    expectEqualTitles(
      {
        ...elementTitles,
        ...attributeTitles,
        ...elementAttributePairTitles
      },
      allHTMLTitles
    );
  });
});
