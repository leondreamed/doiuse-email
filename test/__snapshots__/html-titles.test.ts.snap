// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`css titles > attribute titles 1`] = `
[
  "align attribute",
  "aria-describedby attribute",
  "aria-hidden attribute",
  "aria-label attribute",
  "aria-labelledby attribute",
  "aria-live attribute",
  "background attribute",
  "cellpadding attribute",
  "cellspacing attribute",
  "dir attribute",
  "height attribute",
  "hidden attribute",
  "lang attribute",
  "loading attribute",
  "popover attribute",
  "required attribute",
  "role attribute",
  "srcset and sizes attributes",
  "target attribute",
  "valign attribute",
  "width attribute",
]
`;

exports[`css titles > element titles 1`] = `
[
  "<abbr> element",
  "<acronym> element",
  "<audio> element",
  "<base>",
  "<bdi> element",
  "<blockquote> element",
  "<body> element",
  "<code> element",
  "<del> element",
  "<dfn> element",
  "<dialog> element",
  "<div> element",
  "<form> element",
  "<h1> to <h6> elements",
  "<hr> element",
  "<img> element",
  "<link> element",
  "<marquee> element",
  "<meter> element",
  "<object> element",
  "<p> element",
  "<picture> element",
  "<pre> element",
  "<progress> element",
  "<rp> element",
  "<rt> element",
  "<ruby> element",
  "<select> element",
  "<small> element",
  "<span> element",
  "<strike> element",
  "<strong> element",
  "<style> element",
  "<table> element",
  "<textarea> element",
  "<ul>, <ol> and <dl>",
  "<video> element",
  "<wbr> element",
  "Embedded <svg> image",
  "HTML5 doctype",
  "HTML5 semantics",
  "Image maps",
  "address",
]
`;

exports[`css titles > element-attribute pair titles 1`] = `
[
  "<button type="reset"> element",
  "<button type="submit"> element",
  "<input type="checkbox"> element",
  "<input type="hidden"> element",
  "<input type="radio"> element",
  "<input type="reset"> element",
  "<input type="submit"> element",
  "<input type="text"> element",
  "Local anchors",
  "mailto: links",
]
`;
